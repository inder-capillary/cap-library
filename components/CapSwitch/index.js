/**
*
* CapSwitch
*
*/

import React from "react";
import PropTypes from 'prop-types';
import { Switch } from "antd";
import "./_capSwitch.scss";
// import styled from 'styled-components';
import ComponentWithLabelHOC from '../assets/HOCs/ComponentWithLabelHOC';
const classNames = require('classnames');


class CapSwitch extends React.Component { // eslint-disable-line react/prefer-stateless-function
  render() {
    const {className, children, ...rest} = this.props;
    return (
      <Switch {...rest} className={classNames("cap-switch-v2", className)}>
        {React.Children.toArray(children)}
      </Switch>
    );
  }
}

CapSwitch.propTypes = {
  children: PropTypes.any,
  className: PropTypes.string,
};

const CapSwitchFinal = ComponentWithLabelHOC(CapSwitch);
export default CapSwitchFinal;
