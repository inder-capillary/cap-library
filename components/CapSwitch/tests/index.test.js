import renderer from 'react-test-renderer';
import React from 'react'; // eslint-disable-line no-unused-vars
import {mount, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import CapSwitch from '../index';

configure({ adapter: new Adapter() });
describe('<CapSwitch />', () => {
  it('CapSwitch renders correctly', () => {
    const tree = renderer.create(<CapSwitch />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('check the props is defined', () => {
    const props = {
      size: 'small',
    };
    const CapSwitchComponent = mount(<CapSwitch {...props} />);
    expect(CapSwitchComponent.prop('size')).toBeDefined();
  });

  it('render size input correctly with small value', () => {
    const props = {
      size: 'small',
    };
    const CapSwitchComponent = mount(<CapSwitch {...props} />);
    expect(CapSwitchComponent.prop('size')).toEqual('small');
  });
});
