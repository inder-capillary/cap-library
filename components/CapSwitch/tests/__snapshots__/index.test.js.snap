// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CapSwitch /> CapSwitch renders correctly 1`] = `
<button
  aria-checked={false}
  className="cap-switch ant-switch"
  defaultChecked={false}
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseUp={[Function]}
  role="switch"
  type="button"
>
  <span
    className="ant-switch-inner"
  />
</button>
`;
